api/src/main/java/io/opentelemetry/context/propagation/Propagators.java

```diff
@@ -0,0 +1,133 @@
+ * {@code Context} can then be used implicitly or explicitly by the OpenTelemetry API.
+ *
+ * <p>Example of usage on the client:
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = OpenTelemetry.getTracer();
+ * void onSendRequest() {
+ *   try (Scope scope = tracer.withSpan(span)) {
+ *     Propagators propagators = OpenTelemetry.getPropagators();
+ *     HttpTextFormat textFormat = propagators.getHttpTextFormat();
+ *
+ *     // Inject the span's SpanContext and other available concerns (such as correlations)
+ *     // contained in the specified Context.
+ *     Map<String, String> map = new HashMap<>();
+ *     httpTextFormat.inject(Context.current(), map, new Setter<String, String>() {
+ *       public void put(Map<String, String> map, String key, String value) {
+ *         map.put(key, value);
+ *       }
+ *     });
+ *     // Send the request including the text map and wait for the response.
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>Example of usage in the server:
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = OpenTelemetry.getTracer();
+ * void onRequestReceived() {
+ *   Propagators propagators = OpenTelemetry.getPropagators();
+ *   HttpTextFormat textFormat = propagators.getHttpTextFormat();
+ *
+ *   // Extract and store the propagated span's SpanContext and other available concerns
+ *   // in the specified Context.
+ *   Context context = textFormat.extract(Context.current(), request, new Getter<String, String>() {
+ *     public String get(Object request, String key) {
+ *       // Return the value associated to the key, if available.
+ *     }
+ *   });
+ *   Span span = tracer.spanBuilder("MyRequest")
+ *       .setParent(context)
+ *       .setSpanKind(Span.Kind.SERVER).startSpan();
+ *   try (Scope ss = tracer.withSpan(span)) {
+ *     // Handle request and send response back.
+ *   } finally {
+ *     span.end();
+ *   }
+ * }
+ * }</pre>
+ *
+ * @since 0.3.0
+ */
+@Immutable
+public interface Propagators {
+
+  /**
+   * Returns a {@link HttpTextFormat} propagator.
+   *
+   * <p>The returned value will be a composite instance containing all the registered {@link
+   * HttpTextFormat} propagators. If none is registered, the returned value will be a no-op
+   * instance.
+   *
+   * @return the {@link HttpTextFormat} propagator to inject and extract data.
+   * @since 0.3.0
+   */
+  HttpTextFormat getHttpTextFormat();
+
+  /**
+   * {@link Builder} is used to construct a new {@code Propagators} object with the specified
+   * propagators.
+   *
+   * <p>This is a example of a {@code Propagators} object being created:
+   *
+   * <pre>{@code
+   * Propagators propagators = DefaultPropagators.builder()
+   *     .addHttpTextFormat(new HttpTraceContext())
+   *     .addHttpTextFormat(new HttpCorrelationContext())
+   *     .addHttpTextFormat(new MyCustomContextPropagator())
+   *     .build();
+   * }</pre>
+   *
+   * @since 0.3.0
+   */
+  interface Builder {
```

Comment:
I think this does not need to be an interface, every implementation of the propagators used has its own way to build.

Reply:
Fair enough, lets go without the `Builder`.